### PR TITLE
Add directory client calls

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

**Review by carabasdaniel — APPROVED** [review]

File: Services/UserService.cs [hunk 1]
```diff
@@ -6,10 +6,11 @@
 using System.Net.Http;
 using System.Text.Json;
 using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Logging.Abstractions;
 using Aserto.TodoApp.Configuration;
```
Comment (BogdanIrimie) [inline Services/UserService.cs]: There are some dependencies we might be able to remove:
```c#
using System.Net.Http;
using System.Text.Json;
using Aserto.TodoApp.Configuration;
using System.Text;
using Aserto.Directory.Common.V2;
```

File: Services/UserService.cs [hunk 2]
```diff
@@ -29,55 +30,27 @@ public UserService(IOptions<DirectoryConfig> config)
                 throw new Exception("Invalid config");
             }
 
-            var insecure = opts.Insecure;
-            var directoryServiceURL = opts.ServiceUrl;
+            var logggerFactory = new NullLoggerFactory();
```
Comment (BogdanIrimie) [inline Services/UserService.cs]: Maybe we should log what happens in the directory client: https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line

File: Services/UserService.cs [hunk 3]
```diff
@@ -29,55 +30,27 @@ public UserService(IOptions<DirectoryConfig> config)
...
-                grpcChannelOptions = new GrpcChannelOptions { HttpHandler = httpHandler };
-            }
-
-            var channel = GrpcChannel.ForAddress(
-                directoryServiceURL,
-                grpcChannelOptions);
-
-            this.directoryReaderClient = new ReaderClient(channel);
+            var optionsInt = Microsoft.Extensions.Options.Options.Create(options);
+            this.directoryClient = new DirectoryAPIClient(optionsInt, logggerFactory);
```
Comment (BogdanIrimie) [inline Services/UserService.cs]: You can omit `this` and just have `directoryClient = new DirectoryAPIClient(optionsInt, logggerFactory);`

File: Services/UserService.cs (hunk 2 above)
Comment (BogdanIrimie) [inline Services/UserService.cs]: Small typo, `logggerFactory` should be `loggerFactory`

File: Services/UserService.cs [hunk 4]
```diff
@@ -29,55 +30,27 @@ public UserService(IOptions<DirectoryConfig> config)
...
-                    Subject = new ObjectIdentifier { Type = "user" },
-                    Object = new ObjectIdentifier { Type = "identity", Key = sub },
-                    Relation = new RelationTypeIdentifier { ObjectType = "identity", Name = "identifier" },
-                }
-            };
-
             try
             {
-                var getRelationResponse = await this.directoryReaderClient.GetRelationAsync(getRelationRequest, metaData);
+                var getRelationResponse = await this.directoryClient.GetRelationAsync(subjectType: "user", objType: "identity", objKey: sub, relationName: "identifier", relationObjectType: "identity");
```
Comment (BogdanIrimie) [inline Services/UserService.cs]: We can omit `this` => `this.directoryClient` can become `directoryClient`

File: Services/UserService.cs [hunk 5]
```diff
@@ -29,55 +30,27 @@ public UserService(IOptions<DirectoryConfig> config)
...
                 if (getRelationResponse.Results.Count == 0)
                 {
                     return new GetUserResponse($"No user with identity: {sub}");
                 }
 
                 var getObjRequest = new GetObjectRequest { Param = getRelationResponse.Results[0].Subject };
```
Comment (BogdanIrimie) [inline Services/UserService.cs]: Is `getObjRequest` still used?

File: Services/UserService.cs [hunk 6]
```diff
@@ -29,55 +30,27 @@ public UserService(IOptions<DirectoryConfig> config)
...
-                var getObjResponse = await this.directoryReaderClient.GetObjectAsync(getObjRequest, metaData);
+                var objType = getRelationResponse.Results[0].Subject.Type;
+                var objKey = getRelationResponse.Results[0].Subject.Key;
+                var getObjResponse = await this.directoryClient.GetObjectAsync(objKey, objType);
```
Comment (BogdanIrimie) [inline Services/UserService.cs]: Same as https://github.com/aserto-demo/todo-dotnet-v2/pull/4/files#r1209038665

**Review by BogdanIrimie — APPROVED** [review]

**Pull request merged**